tests/phpunit/tests/l10n/wpTranslationController.php

```diff
@@ -361,4 +361,68 @@ public function test_switch_to_locale_translations_stay_loaded_custom_textdomain
 		$this->assertSame( 'Este es un plugin dummy', $actual );
 		$this->assertSame( 'This is a dummy plugin', $after );
 	}
+
+	/**
+	 * @covers ::has_translation
+	 */
+	public function test_has_translation_with_existing_translation() {
+		load_textdomain( 'wp-tests-domain', DIR_TESTDATA . '/pomo/simple.mo' );
+		$this->assertTrue( WP_Translation_Controller::get_instance()->has_translation( 'baba', 'wp-tests-domain', 'en_US' ) );
+	}
+
+	/**
+	 * @covers ::has_translation
+	 */
+	public function test_has_translation_with_no_translation() {
+		$this->assertFalse( WP_Translation_Controller::get_instance()->has_translation( 'Goodbye', 'wp-tests-domain', 'en_US' ) );
+	}
+
+	/**
+	 * @covers ::has_translation
+	 */
+	public function test_has_translation_with_different_textdomain() {
+		load_textdomain( 'wp-tests-domain', DIR_TESTDATA . '/pomo/simple.mo' );
+		$this->assertFalse( WP_Translation_Controller::get_instance()->has_translation( 'baba', 'custom-domain', 'en_US' ) );
+	}
+
+	/**
+	 * @covers ::has_translation
+	 */
+	public function test_has_translation_with_different_locale() {
+		switch_to_locale( 'fr_FR' );
+		load_textdomain( 'wp-tests-domain', DIR_TESTDATA . '/pomo/simple.mo' );
+		$this->assertFalse( WP_Translation_Controller::get_instance()->has_translation( 'baba', 'wp-tests-domain', 'fr_FR' ) );
```

Comment:
Why would that be false when you switch to fr_FR and then load translations in it? 🤔

Reply:
@swissspidy I couldn't remember my thinking here, but I think this test passes because `simple.mo` is not associated with `fr_FR`. If you think it doesn't make sense, let me know and I'll remove it.

Reply:
Amazing, thanks!